src/duration.js

```diff
@@ -725,6 +729,9 @@ export default class Duration {
 
   /**
    * Return the length of the duration in the specified unit.
+   *
+   * Note as this uses {@link Duration#shiftTo} internally, it can cause the unexpected effects as described there.
```

Comment:
Instead, this should say something like:

> Uses {@link Duration#shiftTo} to convert to the request unit. See the documentation there for details.

The reason this is better is that "unexpected" depends on your expectations. We don't want a _warning_ message; we want a _help you understand_ message.